### DISCUSSION TITLE
Is there a way to get a list of types already loaded from the schema?

### DISCUSSION THREAD

**Discussion by abdullahseba** [webonyx/graphql-php#935] [Q&A]
I am trying to load types that have not already been loaded in the PHP-defined schema so that they can be used in the schema language.  The problem is, `setTypes()` doesn't accept any types which have already been loaded in the schema and `setTypeLoader()` requires all types including those defined in the schema language.  There doesn't seem to be a way of either extracting the types from the schema language for `setTypeLoader()` or a way of getting a list of already loaded types to be excluded in `setTypes()`.  

Any ideas for this?

**Comment by spawnia** [comment]
> There doesn't seem to be a way of either extracting the types from the schema language

There sure is. This library has both higher level utilities to do it (`GraphQL\Utils\BuildSchema`), as well as lower level primitives (`GraphQL\Language\Parser` + building types of the AST nodes).
